src/bit-systems/video-system.ts

```diff
@@ -1,126 +1,69 @@
-    const isPannerNode = isPositionalAudio(audio);
-
-    // TODO this needs more testing
-    if (!isPannerNode && settings.audioType === AudioType.PannerNode) {
-      swapAudioType(world, audioSystem, eid, PositionalAudio);
-    } else if (isPannerNode && settings.audioType === AudioType.Stereo) {
-      swapAudioType(world, audioSystem, eid, StereoAudio);
-    }
-
-    applySettings(audio, settings);
-    removeComponent(world, AudioSettingsChanged, eid);
-  });
-}
+const video2audio = new Map<number, number>();
 
 const OUT_OF_SYNC_SEC = 5;
 const networkedVideoQuery = defineQuery([NetworkedVideo]);
 const mediaVideoQuery = defineQuery([MediaVideo]);
 const mediaVideoEnterQuery = enterQuery(mediaVideoQuery);
+const mediaVideoExitQuery = exitQuery(mediaVideoQuery);
+const mediaPlaybackUpdated = defineQuery([MediaVideo, MediaVideoPlaybackChanged]);
+const mediaLoadStatusQuery = defineQuery([MediaVideo, MediaLoaded]);
+const mediaLoadedQuery = enterQuery(mediaLoadStatusQuery);
+const mediaUnloadedQuery = exitQuery(mediaLoadStatusQuery);
 export function videoSystem(world: HubsWorld, audioSystem: AudioSystem) {
-  mediaVideoEnterQuery(world).forEach(function (eid) {
-    const videoObj = world.eid2obj.get(eid) as Mesh;
+  mediaVideoEnterQuery(world).forEach(function (videoEid) {
+    const videoObj = world.eid2obj.get(videoEid) as Mesh;
     const video = (videoObj.material as MeshStandardMaterial).map!.image as HTMLVideoElement;
-    if (MediaVideo.autoPlay[eid]) {
+    if (MediaVideo.autoPlay[videoEid]) {
       video.play().catch(() => {
         // Need to deal with the fact play() may fail if user has not interacted with browser yet.
         console.error("Error auto-playing video.");
       });
     }
-    const audio = world.eid2obj.get(makeAudioSourceEntity(world, video, audioSystem))!;
+    const audioEid = makeAudioSourceEntity(world, video, audioSystem);
+    video2audio.set(videoEid, audioEid);
+    const audio = world.eid2obj.get(audioEid)!;
     videoObj.add(audio);
     // Note in media-video we call updateMatrixWorld here to force PositionalAudio's updateMatrixWorld to run even
     // if it has an invisible parent. We don't want to have invisible parents now.
   });
-
-  audioEmitterSystem(world, audioSystem);
-
+  mediaLoadedQuery(world).forEach(videoEid => {
+    const audioEid = video2audio.get(videoEid)!;
+    const rootEid = MediaLoaded.rootRef[videoEid];
+    if (APP.audioOverrides.has(rootEid)) {
+      const audioSettings = APP.audioOverrides.get(rootEid)!;
+      APP.audioOverrides.set(audioEid, audioSettings);
+    }
+  });
+  mediaUnloadedQuery(world).forEach(videoEid => {
+    const audioEid = video2audio.get(videoEid)!;
+    APP.audioOverrides.delete(audioEid);
+  });
+  mediaVideoExitQuery(world).forEach(videoEid => {
+    const rootEid = MediaLoaded.rootRef[videoEid];
+    APP.audioOverrides.delete(rootEid);
+    video2audio.delete(videoEid);
+  });
+  mediaPlaybackUpdated(world).forEach(videoEid => {
```

Comment:
Rather than adding the component, querying for it, then changing APP.isAudioPaused, we could do the last part directly where we are toggling the video. APP.isAudioPaused is basically a "proto-tag-component" anyway

Reply:
I think the second part of your comment is related to what I mentioned in my previous comment. Regarding the first part, the main reason to send that "event" is that there is a `video2audio` association map that lives in the video system. But I think this is a good application of the escape hatch pattern. This would live in a `MediaVideo` component map and the we can get the `AudioEmitter` and update the flags from anywhere. I'm going to push that.